gradle-android-command-plugin/src/main/groovy/com/novoda/gradle/command/Monkey.groovy

```diff
@@ -6,30 +6,49 @@ class Monkey extends AdbTask {
 
     def events
     def seed
+    def categories
 
     protected handleCommandOutput(def text) {
         super.handleCommandOutput(text)
-        if (text.contains("Monkey aborted"))
+        if (text.contains("Monkey aborted")) {
             throw new GroovyRuntimeException('Monkey run failed')
+        }
     }
 
     @TaskAction
     void exec() {
-        def arguments = ['shell', 'monkey', '-p', packageName, '-v', getEvents()]
-        if (getSeed())
+        def arguments = ['shell', 'monkey']
+        arguments += ['-p', packageName]
+        arguments += getFormattedCategories()
+        arguments += ['-v', getEvents()]
+        if (getSeed()) {
             arguments += ['-s', getSeed()]
+        }
         assertDeviceAndRunCommand(arguments)
     }
 
+    def getFormattedCategories() {
+        getCategories().collect { "-c " + it }
```

Comment:
Since every category needs to be prefixed with -c , we came up with a solution to make a separate method for this. 
I can inline this method but I am not sure if this will be clearer. 
Do you have any other suggestions ?